src/drupal8/docs/install/multi-site.md

```diff
@@ -0,0 +1,171 @@
+# Setting up a multi-site in Workspace
+
+First you will need to follow the steps to set up a Workspace environment for either a [new project] or [existing project], which ever is relevant to your scenario.
+
+This guide assumes that you have an understanding of how to set Drupal up for multi-site use. More information can be found in the [official multi-site documentation](https://www.drupal.org/docs/8/multisite).
+
+## Contents
+* [Setting up the hostnames](#define-the-hostnames)
+* [Create the individual databases](#creating-multiple-databases-during-install)
+  * [Importing databases during install](#importing-databases-during-install)
+* [Update sites.php](#update-sitesphp)
+* [Update the Drupal settings]
+* [Add drush aliases](#add-drush-aliases)
+* [Allow write access for CI](#allow-directory-write-access-for-ci)
+
+
+## Define the hostnames
+At this point you should have at least a `workspace.yml` file in the project root.  
+Here the hostnames for the multi-sites can be defined using the `hostname_aliases` attribute:
+```yaml
+attribute('hostname_aliases'):
+  - site-one
+  - site-two
+  - site-three
+```
+
+These will have the `.my127.site` suffix added and will end up as:
+* site-one.my127.site
+* site-two.my127.site
+* site-three.my127.site
+
+
+## Creating multiple databases during install
+_If you are using table prefixes rather than multiple databases then you can skip this section._
+
+In order to use multiple databases in single mysql container you will need to change the defaults to use the continuous pipe mysql image.
+
+In `workspace.yml` add:
+```yaml
+attribute('mysql.image'): quay.io/continuouspipe/mysql5.7
+attribute('mysql.tag'): latest
```

Comment:
@kierenevans, I've left this as `latest` for now. Will the continouspipe image be updated or should I change this to either use `stable` or add the grants in the install steps?

Reply:
Let's update to use the init step grants instead of the CP image

Reply:
Sure - my concern with that approach is that they don't trigger once you have a DB dump in `tools/assets`. Probably not an issue for the initial install, but if someone destroys and re-installs then the databases won't be created.

Taking a look at other options, would it work if we created them in the `before('harness.install')` event?

Reply:
I think it's triggered too early to use against a running container: https://github.com/my127/workspace/blob/0.1.x/src/Types/Workspace/Installer.php#L79

The `ws enable` command is called by the `after.harness.install` event triggered here:
https://github.com/my127/workspace/blob/0.1.x/src/Types/Workspace/Installer.php#L97
and received here: https://github.com/inviqa/harness-base-php/blob/17d88d3ba9a48b1715211ee81307284d70914ce7/src/_base/harness/config/events.yml#L2-L4

Only then is the set of docker containers brought up.

After that we are left with either build step, install step, init step, or customising `enable.sh` to be able to do extra things in between.

Reply:
> Sure - my concern with that approach is that they don't trigger once you have a DB dump in `tools/assets`.
I think that's why we did the grants as the init step rather than install step.

Reply:
Updated to use the init steps.